sentry-android-core/src/main/java/io/sentry/android/core/UserInteractionIntegration.java

```diff
@@ -119,6 +125,10 @@ public void register(@NotNull IHub hub, @NotNull SentryOptions options) {
             "UserInteractionIntegration enabled: %s",
             this.options.isEnableUserInteractionBreadcrumbs());
 
+    this.performanceEnabled =
+        options.isTracingEnabled()
+            && ((SentryAndroidOptions) options).isEnableUserInteractionTracing();
```

Comment:
Why do you need to cast to `SentryAndroidOptions`? `register` already does this casting.

Reply:
mm, IDE/build was complaining so I had to do it. But I think I will remove it altogether in favour of your suggestion below (just checking the options directly instead of having this obscure boolean flag)

Reply:
Likely because you were using `options` instead of `this.options`, after casting.

Reply:
indeed